### ISSUE TITLE
3.5 b1 no longer working in Xcode 12

### ISSUE THREAD

**Issue by sammcewan** [sumup/sumup-ios-sdk#98]
We need movement on this. 

Xcode 12 will no longer build our app when using the version with UIWebView removed, master build works fine. We see this as a key risk for us.

```
ld: in /Pods/SumUpSDK/SumUpSDK.embeddedframework/SumUpSDK.framework/SumUpSDK(AudioReaderSDK.o), building for iOS Simulator, 
but linking in object file built for free standing, file Pods/SumUpSDK/SumUpSDK.embeddedframework/SumUpSDK.framework/SumUpSDK' for architecture x86_64
```

Easily reproducible by trying to build and run example app on Xcode b4.

Can we get 3.5 marked as production ready (it's been nearly 3 months) and this error remediated ASAP?

**Comment by Mike-Lopez** [comment]
Hi,

Our apologies for the delay. I have created a ticket with our Product team and escalated this to be looked at and have a fix deployed asap.

**Comment by sammcewan** [comment]
@Mike-Lopez thanks. This is still the case with Xcode 12 b5 which was released this morning.

Related to https://github.com/sumup/sumup-ios-sdk/issues/79

**Comment by Bas94** [comment]
When can we get out of beta into an official 3.5 release? This issue is holding us back from updating our build pipeline with newer dependencies. Please fix this ASAP.

**Comment by sammcewan** [comment]
@Mike-Lopez We're about to pull the trigger on a project to remove SumUp from our point of sale as a result of this. Can we please have an update. We're getting desperate.

**Comment by jahenry** [comment]
@sammcewan @Bas94 sincere apologies for the delayed feedback and the inconvenience!  Understand your concerns, the team is working on a new release as we speak and it's scheduled to be released by end of week!  Sorry again for the delay.  We'll keep you posted.
Kind regards
James

**Comment by mollidor** [comment]
Hi all.

We just published [v4.0.0-beta.1](https://github.com/sumup/sumup-ios-sdk/releases/tag/v4.0.0-beta.1) . Check the [Migration Guide](https://github.com/sumup/sumup-ios-sdk/blob/develop/MIGRATION.md#sumup-mpos-sdk-400-beta1-migration-guide), integration of the SDK should have become way easier.

Kind regards,
Lukas

**Comment by sammcewan** [comment]
@mollidor Looking good but we need the bundle version to follow apple's conventions of a period separated list currently it is 4.0.0-beta.1 which is not able to be submitted. I've changed this manually in the xcframework plists but would be good to get this updated so others don't have this issue. Thanks.

**Comment by mollidor** [comment]
Hi @sammcewan 

Thanks for your message. We have successfully uploaded a sample app with `4.0.0.beta.1` but then decided to follow the pattern [recommended by semver](https://semver.org/#spec-item-90). Seems we should have asked ASC for an opinion as well.

Can you confirm that `4.0.0-beta.1` will be rejected by ASC validation and changing the bundle version manually to `4.0.0.beta.1` will make it pass validation? Which version in which plist did you change exactly? Would you mind posting the message here? There's multiple references to this bundle version and I want to make sure I fully understand.

Appreciate your help.

Kind regards,
Lukas

**Comment by sammcewan** [comment]
@mollidor Here's the error in its entirety.

> ERROR ITMS-90060: "This bundle is invalid. The value for key CFBundleShortVersionString '4.0.0-beta.1' in the Info.plist file at '${bundlePath}' must be a period-separated list of at most three non-negative integers. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring"
Return status of iTunes Transporter was 1: ERROR ITMS-90060: "This bundle is invalid. The value for key CFBundleShortVersionString '4.0.0-beta.1' in the Info.plist file at '${bundlePath}' must be a period-separated list of at most three non-negative integers. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring"
The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.

I modified by editing these plists:

> Pods/SumUpSDK/SumUpSDK.xcframework/ios-i386_x86_64-simulator/SumUpSDK.framework/Info.plist
> Pods/SumUpSDK/SumUpSDK.xcframework/ios-armv7_arm64/SumUpSDK.framework/Info.plist

**Comment by mollidor** [comment]
Thanks a lot. So changing it to `4.0.0.beta.1` fixed it for you? Happy to do so, just making sure that this would pass ASC validation which officially is more strict:

> This key is a user-visible string for the version of the bundle. The required format is three period-separated integers, such as 10.14.1. The string can only contain numeric characters (0-9) and periods.

**Comment by sammcewan** [comment]
@mollidor  I changed it to 4.0.0 unsure if 4.0.0.beta.1 would pass.

**Comment by mollidor** [comment]
Hi all.

We published [v4.0.0](https://github.com/sumup/sumup-ios-sdk/releases/tag/v4.0.0) .
Going forward we will make sure to not add preview specifiers as a suffix to the bundle version.

Best,
Lukas
